docs/source/basics/tokenizers.rst

```diff
@@ -0,0 +1,239 @@
+.. _tokenizers_usage_label:
+
+==========
+Tokenizers
+==========
+
+Tokenizers are a key component of any LLM. They convert raw text into token IDs, which index into embedding vectors that are
+understood by the model.
+
+In torchtune, tokenizers play the role of converting Messages object into token IDs and any necessary model-specific special tokens.
+They use byte-pair encoding algorithms to convert raw strings into integer IDs. These are usually based on SentencePiece or TikToken.
+
+Downloading tokenizers from Hugging Face
+----------------------------------------
+
+Models hosted on Hugging Face are also distributed with the tokenizers they were trained with. These are automatically downloaded alongside
+model weights when using ``tune download``. For exmple, this command downloads the Mistral-7B model weights and tokenizer:
+
+.. code-block:: bash
+
+    tune download mistralai/Mistral-7B-v0.1 --output-dir /tmp/Mistral-7B-v0.1 --hf-token <HF_TOKEN>
+    cd /tmp/Mistral-7B-v0.1/
+    ls tokenizer.model
+    # tokenizer.model
+
+Loading tokenizers from file
+----------------------------
+
+Once you've downloaded the tokenizer file, you can load it into the corresponding tokenizer class by pointing
+to the file path of the tokenizer model in your config or in the constructor. You can also pass in a custom file path if you've already
+downloaded it to a different location.
+
+.. code-block:: python
+
+    # In code
+    from torchtune.models.mistral import mistral_tokenizer
+
+    m_tokenizer = mistral_tokenizer("/tmp/Mistral-7B-v0.1/tokenizer.model")
+    type(m_tokenizer)
+    # <class 'torchtune.models.mistral._tokenizer.MistralTokenizer'>
+
+.. code-block:: yaml
+
+    # In config
+    tokenizer:
+      _component_: torchtune.models.mistral.mistral_tokenizer
+      path: /tmp/Mistral-7B-v0.1/tokenizer.model
+
+Setting max sequence length
+---------------------------
+
+Setting max sequence length can give you control over memory usage and adhere to model specifications.
+
+.. code-block:: python
+
+    # In code
+    from torchtune.models.mistral import mistral_tokenizer
+
+    m_tokenizer = mistral_tokenizer("/tmp/Mistral-7B-v0.1/tokenizer.model", max_seq_len=8192)
```

Comment:
Give examples.. set a really small max seq len and tokenize something then show the results